x-pack/plugins/security_solution/server/endpoint/services/agent/clients/lib/types.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ResponseActionAgentType,
+  ResponseActionsApiCommandNames,
+} from '../../../../../../common/endpoint/service/response_actions/constants';
+import type { HostStatus } from '../../../../../../common/endpoint/types';
+
+export interface RawSentinelOneInfo {
+  sentinel_one: {
+    agent: {
+      uuid: string;
+      last_active_date: string;
+      network_status: string;
+      is_active: boolean;
+      is_pending_uninstall: boolean;
+      is_uninstalled: boolean;
+    };
+  };
+}
+
+export interface AgentStatuses {
+  [agentId: string]: {
+    agentId: string;
+    agentType: ResponseActionAgentType;
+    capabilities: string[];
+    found: boolean;
+    isolated: boolean;
+    isPendingUninstall: boolean | undefined;
+    isUninstalled: boolean | undefined;
```

Comment:
We use them for sentinel one to determine if we should allow isolate/unisolate actions on alerts. If either is true then the actions should be disabled.

Reply:
They should be removed since they are sentinelone specific. Is there another property we can use? perhaps the `status` value?

If we are going to include `agentType` specific data, then we should discuss where that data should go so that we don't blow up this response with all kinds of data that are agent type specific as we move forward.

Reply:
I see you point. I have a couple of questions regarding this. Like, I'm not sure why this logic was implemented in the first place, i.e. to not allow isolate/release actions on uninstalled/uninstalling sentinelOne agents.  If we use `status` as a check to enable isolate/release action would this mean that we use it only for sentinelOne. If yes, that would mean that these actions work slightly differently than endpoint agents. Is that something a user need to be aware of?

What do you think?

Reply:
We have code today that checks for the uninstall/uninstalled? can you point me to it?

Reply:
Here https://github.com/elastic/kibana/blob/c0e7e1b14008fe94ac3bc82114582c2f27b3acdf/x-pack/plugins/security_solution/public/detections/components/host_isolation/use_host_isolation_action.tsx#L130.

It was first introduced in github.com/elastic/kibana/pull/168441

Reply:
I'll try and remove the capabilities though. I agree it doesn't make sense to capture this in this API.

Reply:
updated to use  unenrolled `status` like you suggested. 1db3d63abef5e53ee95a556e4a5fb9e8c332bf24